python/ql/lib/semmle/python/frameworks/Stdlib.qll

```diff
@@ -4230,7 +4230,7 @@ module StdlibPrivate {
         preservesValue = true
       )
       or
-      input = "Argument[0]" and
+      input = "Argument[0..]" and
```

Comment:
With the change from https://github.com/github/codeql/pull/17566/commits/315890680cc38d86c49c37e18d104fee2cb95775, shouldn't this also be reverted back to `Argument[0]`?

```suggestion
      input = "Argument[0]" and
```

Reply:
No, but there should probably be a comment above each of these cases. This particular case is handling imprecise taint going from an input to the entire returned collection; that should also happen for the later arguments (which do not have to be explicit keyword arguments but could be just a tainted dictionary).

Reply:
I think that raw dict comprehensions do not actually work: https://github.com/github/codeql/blob/main/python/ql/test/library-tests/dataflow/coverage/test.py#L182-L184

Reply:
I added some explanatory comments. Also the link to the function doc was missing, so I had to hunt it down...

Reply:
> that should also happen for the later arguments (which do not have to be explicit keyword arguments but could be just a tainted dictionary).

I beg to differ. 

```
$ python -c 'dict([("k1", 1)], [("k2", 2)])'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: dict expected at most 1 argument, got 2
```

Reply:
You have to use the right syntax, though:
```
>>> kwargs = {"a": 1, "b": 2}
>>> dict([("c",2)], **kwargs)
{'c': 2, 'a': 1, 'b': 2}
```

Reply:
But maybe we can target this more precisely by asking for the dictsplat argument?

Reply:
> You have to use the right syntax, though:
> 
> ```
> >>> kwargs = {"a": 1, "b": 2}
> >>> dict([("c",2)], **kwargs)
> {'c': 2, 'a': 1, 'b': 2}
> ```

yes, but those are no longer positional arguments, so should not be matched by `Argument[0..]`. `dict(list, a=42, b=43)` also works, but those again, those are keyword arguments, and would not be matched by `Argument[0..]`

Reply:
Ah, of course, you are right! :-)